### PR TITLE
29 add from state

### LINKED ISSUE #29: Add from_state. Takes list of qubits and creates state

### PR DESCRIPTION
**Pull request by JonatanMentzer**
Resolves #29 

from_state or new_qubits as it is called in the code can create any state of valid qubits.

### CODE REVIEW

File: src/register.rs [hunk 1]
```diff
@@ -50,6 +51,64 @@ impl Register {
...
+    /// Two qubits looks like this: [array![[Complex::new(1.0, 0.0)], [Complex::new(0.0, 0.0)]], 
+    ///                              array![[Complex::new(0.0, 0.0)], [Complex::new(1.0, 0.0)]]]
+    /// 
+    /// **Panics** if 2 dimensional array doesn't contain 2 elements
+    /// or if their probability doesn't add to 1
+    pub fn new_qubits(input_bits: &[Array2<Complex<f64>>]) -> Self {
+        Self::try_new_qubits(input_bits).expect("Incorrect input qubits")
+    }
+
+    pub fn try_new_qubits(input_bits: &[Array2<Complex<f64>>]) -> Result<Self, OperationError> {
```
Comment (simonengstrm) [inline src/register.rs:67]: Would be nice to have some documentation of this function

File: src/register.rs [hunk 2]
```diff
@@ -50,6 +51,64 @@ impl Register {
...
+            .iter()
+            .fold(base_state, |a, b| linalg::kron(&b, &a));
+
+        return Ok(Self {
+            state: state_matrix,
+            size: input_bits.len(),
+        });
+    }
+
+    /// Checks if input qubit has total probability 1
```
Comment (simonengstrm) [inline src/register.rs:92]: Add docs note about parameter/return?

File: src/register.rs [hunk 3]
```diff
@@ -50,6 +51,64 @@ impl Register {
...
+            total_prob += values.norm_sqr();
+        }
+        //Chech if total prob is 1
+        if total_prob.abs() < 0.000001 {
+            return true;
+        }
+        return false;
+    }
+
+    /// Checks that input qubit is correct
```
Comment (simonengstrm) [inline src/register.rs:106]: Add docs note about parameter/return?

**Review by simonengstrm — CHANGES_REQUESTED** [review]
Looks good! Would be nice with some more docs in a couple of places.
Also maybe squash commit message history when merging

**Comment by BAMF0** [comment]
We should make sure to keep commits in imperative form to be consistent!

**Review by BAMF0 — COMMENTED** [review]
I agree with the changes as requested by @simonengstrm

**Comment by JonatanMentzer** [comment]
Will squash and merge after PR is accepted.

**Review by simonengstrm — APPROVED** [review]
lgtm!

**Pull request merged**